zbuf/encoder.go

```diff
@@ -0,0 +1,46 @@
+package zbuf
+
+import (
+	"reflect"
+
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type Encoder struct {
```

Comment:
I don't think we need this. It doesn't act like, say, a json.Encoder, because it has the internal loop over slice-like types, and the json.Encoder functionality for setting output options we can get by extending the resolver.MarshalContext.

Reply:
I get the point about looping through slices as a user might want to have an array serialized (though this was convenient!). Will remove this feature.

As far as removing this, I still think this is a nice convenience utility that needs to exist in zbuf since resolver (a dependency of zbuf) has no notion of a zbuf.Reader/Writer. Without this (and adjusted to remove the slice reflection) the code in ls would look something like this:

```golang
rec, err := marshalcontext.MarshalRecord(v)
if err != nil {
    return err
}
if err := zwriter.Write(rec); err != nil {
    return err
}
```

Compared to:
```golang
if err := encoder.Encode(v); err != nil {
    return err
}
```

The go standard library saw it fit to include this convenience. Perhaps change the name to: `zbuf.MarshalWriter`?

Perhaps what I'm struggling with is that having Marshal in resolver strains some of the flexibility I'm looking for. Thoughts?

Reply:
After I removed the reflection piece of Encoder the amount of code saved for having another struct in zbuf made it seem not worth keeping in light the objections stated. I did you as suggested.

Still think it would nice to have something that encoded a struct directly into a write stream, perhaps this is something we can keep in mind as the design of resolver continues to evolve.